cocotb/__init__.py

```diff
@@ -191,18 +191,18 @@ def _sim_event(level, message):
     from cocotb.result import TestFailure, SimFailure
 
     if level is SIM_TEST_FAIL:
-        scheduler.log.error("Failing test at simulator request")
-        scheduler.finish_test(TestFailure("Failure from external source: %s" %
-                              message))
+        cocotb.log.error("Failing test at simulator request")
+        if cocotb.regression_manager is not None:
```

Comment:
This can occur if the regression_manager hasn't finished being created when the test fails (`_initialize_testbench` exists early). This won't be a problem i teh future after I add the ability to register this callback handler late (last operation in init).

Reply:
Added comment here.